packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -744,6 +745,18 @@ class AppBar extends StatefulWidget implements PreferredSizeWidget {
   /// {@endtemplate}
   final bool forceMaterialTransparency;
 
+  /// {@template flutter.material.appbar.customizeSemanticsSortKeys}
+  /// Whether to customize the sort key behavior of the app bar.
+  ///
+  /// If this is set to false, the app bar will use the default sort key behavior,
+  /// which means the flexible space will appear after the main content in the semantics tree.
+  ///
+  /// Set this to true if you want to customize the semantics sort keys of the app bar.
+  ///
+  /// Defaults to false.
+  /// {@endtemplate}
+  final bool customizeSemanticsSortKeys;
```

Comment:
This needs a better name. maybe called `useDefaultSemanticsOrder` and doc should also comment what the default semantics order and how they can change it (by assigning the sortkey in the input widget).

and also should add a `See also` that points to semanticsSortKey doc